bin/propolis-server/src/lib/initializer.rs

```diff
@@ -498,15 +543,15 @@ impl<'a> MachineInitializer<'a> {
                             slog::o!("component" => format!("nvme-{}", name)),
                         ),
                     );
-                    let id =
-                        self.inv.register_instance(&nvme, bdf.to_string())?;
-                    let _ = self.inv.register_child(child, id).unwrap();
+                    self.devices
+                        .insert(format!("pci-nvme-{}", bdf), nvme.clone());
```

Comment:
tiny, unimportant nit: this could be
```suggestion
                        .insert(format!("pci-nvme-{bdf}"), nvme.clone());
```

Reply:
Done.